contracts/USSLib.sol

```diff
@@ -0,0 +1,113 @@
+        // "committed relayer" address.
+        address relayer;
+        // Token that is deposited on origin chain by depositor.
+        address inputToken;
+        // Token that is received on destination chain by recipient.
+        address outputToken;
+        // The amount of input token deposited by depositor.
+        uint256 inputAmount;
+        // The amount of output token to be received by recipient.
+        uint256 outputAmount;
+        // Origin chain id.
+        uint256 originChainId;
+        // Destination chain id.
+        uint256 destinationChainId;
+        // The id uniquely identifying this deposit on the origin chain.
+        uint32 depositId;
+        // The timestamp on the destination chain after which this deposit can no longer be filled.
+        uint32 expiryTimestamp;
+        // Data that is forwarded to the recipient.
+        bytes message;
+    }
+
+    struct SlowFill {
+        RelayData relayData;
+        RelayExecutionType executionType;
+        int256 payoutAdjustmentPct;
+    }
+
+    // Contains information about a relay to be sent along with additional information that is not unique to the
+    // relay itself but is required to know how to process the relay. For example, "updatedX" fields can be used
+    // by the relayer to modify fields of the relay with the depositor's permission, and "repaymentChainId" is specified
+    // by the relayer to determine where to take a relayer refund, but doesn't affect the uniqueness of the relay.
+    struct RelayExecution {
+        RelayData relay;
+        bytes32 relayHash;
+        uint256 updatedOutputAmount;
+        address updatedRecipient;
+        bytes updatedMessage;
+        uint256 repaymentChainId;
+        RelayExecutionType executionType;
+        int256 payoutAdjustmentPct;
+    }
+
+    struct RelayExecutionInfo {
+        address recipient;
+        RelayExecutionType executionType;
+        uint256 outputAmount;
+        int256 payoutAdjustmentPct;
+        bytes message;
+    }
+
+    // TODO: I know within Solidity we can name this event the same as SpokePool.FundsDeposited but does that make
+    // it harder at the typescript level to distinguish the events?
+    event FundsDeposited(
+        uint256 indexed originChainId,
+        uint256 destinationChainId,
+        uint256 inputAmount,
+        uint256 outputAmount,
+        uint32 indexed depositId,
+        uint32 quoteTimestamp,
+        uint32 expiryTimestamp,
+        address inputToken,
+        address outputToken,
+        address indexed depositor,
+        address recipient,
+        address relayer,
+        address depositRefundCallbackAddress,
+        bytes message
+    );
+
+    event FilledRelay(
+        uint256 inputAmount,
+        uint256 outputAmount,
+        uint256 repaymentChainId,
+        uint256 indexed originChainId,
+        uint256 destinationChainId,
+        uint32 indexed depositId,
+        uint32 fillDeadline,
```

Comment:
Assume this is a 1:1 copy of `expiryTimestamp` ?

Reply:
it is yeah i wasn't sure what to name it or whether to keep them all the same name, any thoughts? In my mind `fillDeadline` == `expiryTimestamp` but the former might be more intuitive